amida-tech/orange-api
# Feature/orange 835 create user scopes

# This will break the register form on the app

# What does this do?
* Support the `programAdministrator` role
* Only allow `admin` or `programAdministrator` *scope* to use create user endpoint
* Forward the received jwt to auth service when calling the auth create user endpoint.
* Include the *role* as a *scope* when creating the user in auth service.
* Gets rid of the legacy token auth we were still using in tests, using actual auth in tests now

# How do I test it?
Setup auth service from an empty db, and take note of the admin password it creates and prints to the  screen.

*Note from Jonah:* You will also need to set the env var `AUTH_SERVICE_REGISTRAR_SCOPES` in the auth service to `programAdministrator`.

Via Postman:

---
 - Login to auth with user "admin" and the password it printed out via postman and get the token.
 - Use the admin token to call the register user endpoint on orange - create a user with the role "programAdministrator"
   - This should succeed
---
 - Login as the "programAdministrator" user you just created on auth and get the token.
 - Use that token to call the register user endpoint on orange - create a user with either the "clinician" or "user" role
   - This should succeed
---
 - Login as the last user you just created on auth and get the token (this user should be either a "clinician" or "user".
 - Use that token to call the register user endpoint on orange.
   - This should fail
---
 - Call the register user endpoint on orange without a token
   - This should fail
